firedrake/assemble.py

```diff
@@ -2185,6 +2265,27 @@ def _as_parloop_arg_layer_count(_, self):
 class _FormHandler:
     """Utility class for inspecting forms and local kernels."""
 
+    @staticmethod
+    def iter_active_coordinates(form, kinfo):
+        """Yield the form coordinates referenced in ``kinfo``."""
+        all_meshes = extract_domains(form)
+        for i in kinfo.active_domain_numbers.coordinates:
+            yield all_meshes[i].coordinates
+
```

Comment:
Note: Make GlocalKernel args and Parloop args only for the active coordinates (and so on).